### ISSUE TITLE
SC2188 suggested fix triggers SC2217

### ISSUE THREAD

**Issue by tripleee** [koalaman/shellcheck#2603]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2188
- My shellcheck version (`shellcheck --version` or "online"): 0.8.0
- [X] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [X] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh
<<EOF
Some human-readable documentation here
EOF
```

#### Here's what shellcheck currently says:

```
<<EOF
^---^ SC2188 (warning): This redirection doesn't have a command.
Move to its command (or use 'true' as no-op).
```

(I split the error message over two lines for legibility, to make sure you see the suggested fix.)

If you change this to

```
true <<EOF
Some human-readable documentation here
EOF
```
`shellcheck` prints
```
true <<EOF
^--------^ SC2217 (warning): Redirecting to 'true', a command that doesn't read stdin.
Bad quoting, wrong command or missing xargs?
```

#### Here's what I wanted or expected to see:

The suggestion should perhaps choose a different workaround than `true`.

I find that using `:` instead of `true` works fine, though perhaps that's not "expected" as such -- this command also does not read its standard input, so perhaps in some sense it should trigger the same warning? `:` also has the pesky side effect of evaluating its arguments (but that's also shared by `true`).

**Comment by koalaman** [comment]
In practice, SC2216 (`foo | ls`) and SC2217 (`ls < foo`) primarily catch bad attempts at echoing/listing/copying/moving things. For no pragmatic reason it also included a pile of other commands that will likely never be involved in a real world case, like `chgrp` and `ulimit`. As you convincingly argue, it's annoying to have true there, so true/false are now removed. Thanks!
